internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -118,6 +130,34 @@ func (hr *HypershiftAgentServiceConfigReconciler) getKubeconfigSecret(ctx contex
 	return secret, nil
 }
 
+func (hr *HypershiftAgentServiceConfigReconciler) ensureSpokeAgentInstallCRDs(ctx context.Context) error {
+	// Fetch agent-install CRDs using in-cluster client
+	localCRDs := &apiextensionsv1.CustomResourceDefinitionList{}
+	listOpts := []client.ListOption{
+		client.MatchingLabels{fmt.Sprintf("operators.coreos.com/assisted-service-operator.%s", hr.Namespace): ""},
+	}
+	if err := hr.Client.List(ctx, localCRDs, listOpts...); err != nil {
+		return pkgerror.Wrap(err, "Failed to list in-cluster CRDs")
+	}
+	if len(localCRDs.Items) == 0 {
+		return pkgerror.New("agent-install CRDs are not available")
+	}
+
+	// Ensure CRDs exist on the spoke cluster
+	for _, crd := range localCRDs.Items {
+		c := crd
```

Comment:
Pretty sure you don't need this here.

Reply:
Added mutate func, so need it now :) I.e. to avoid: "Using the variable on range scope `crd` in function literal"